Convert issue tempates to forms

gravityfargo opened the pull request:
While digging through issues it became apparent to me that tracking supported hardware is confusing as it stands. I created a new device table where we can track specific device support by SKU in addition to the families and models which already kind of exists on the current supported devices page. The issue templates have been edited to reflect the need for a SKU.

The whole SKU thing: If I understand correctly each model will have its own Device ID set by corsair at the factory. It makes sense to me that the SKU is the next logical identifier. Nomenclatures on Corsair devices are agrevating, I don't think it's a great system to base "Is this a duplicate issue/device request?" questions on. User reporting isn't awesome either unless it's required that they have to go out of their way to look for the physical model number sticker. I think the documentation is pretty straightforward in the new table. I bolded one keyboard and one mouse model already because I own them, and they work flawlessly. Their device id was inputted as well.

I've prettied up the readme a bit. The Github issues are messy, and I assume the google group isn't even monitored let alone moderated, so I removed that link. It's been two years since macOS support dropped and there's a pinned issue so I removed that. Moved the disclaimer to the bottom with additions, and added "Corsair" in the name at the top. "RGB Driver" isn't representative of the project IMO. Supported devices got their own section, broken down by device type. I moved the contact info around and removed its section because redundant.

 I overhauled the Issue templates to accommodate the SKU thing and make more sense instruction-wise. There are 4 issue templates proposed, each with instructions on how to title, what information is required, and a copy-paste template for ease of use. I like standardization :) 

Sorry for a billion commits in this pull request for small things, I'm still new at git and whatnot. I just did what made sense to finish my idea. 

I posted a message in the IRC about discussing this prior to submitting a pull request, but with timezones and not having the time to monitor the chat I'll just answer questions here.

Ravenslofty commented on README.md:
```diff
@@ -16,33 +15,46 @@ Major features:
-* [Matrix](https://matrix.to/#/#ckb-next:matrix.org): `#ckb-next:matrix.org`
-* Mailing list: [ckb-next@googlegroups.com](https://groups.google.com/forum/#!forum/ckb-next)
-* [GitHub Issues](https://github.com/ckb-next/ckb-next/issues)
+***Disclaimer:*** ckb-next is not an official Corsair product. It is not supported by, affiliated with, or endorsed by Corsair. It is licensed under the GNU General Public License (version 2) in the hope that it will be helpful, but with NO WARRANTY of any kind.
```
The disclaimer was put near the top of the readme for a reason, please do not bury it.

gravityfargo replied:
It will be moved back in my next push.

gravityfargo reviewed (commented):
So there are a lot of discrepancies between usb.h and the actual nomenclatures found on Corsair's product pages. I had added comments to my local usb.h but didn't feel its in the scope of this PR to address them. For instance: All of the entries in this supported_hardware file that doesn't have a Device ID do not have a 1:1 match in usb.h that can be verified. The SABRE mouse models are supported via usb.h in order of sensor type (laser, optical, etc) but according to Corsair's product pages, those features do not differentiate the "RGB, RGB Champion, and RGB Champion Ultra-Lite" I still included those modes as being supported in this new matrix in accordance with the current supported devices list wiki. I feel this is import to address for general users to have a straightforward supported devices list to reference.

Ravenslofty commented:
> The SABRE mouse models are supported via usb.h in order of sensor type (laser, optical, etc) but according to Corsair's product pages, those features do not differentiate the "RGB, RGB Champion, and RGB Champion Ultra-Lite"

`1b14`, `1b19` and `1b2f` are all SABRE RGB-branded, as I understand it. I'm not actually sure if the Champion series are supported, but maybe @tatokis knows.

gravityfargo commented:
### Change log
- Issue templates are now forms. 
  - See the new interface by [submitting a test issue ](https://github.com/gravityfargo/ckb-next/issues/new/choose) in my fork.
  - I have submitted one of each request already to see [the result](https://github.com/gravityfargo/ckb-next/issues) of filling in the form.
  - the config.yml is necessary to enable the form functionality. 
  - I added a new issue form for people to request help. Since I disabled the ability to create blank issues (in config.yml) I figured we needed a catch-all form. 
- [Supported Devices](https://github.com/gravityfargo/ckb-next/blob/master/Supported_Hardware.md) list has been elaborated on and formatted as a table. 
- The readme got updated
  - I moved stuff around based on my preference. Not necessary tbh.
  - The [Supported Devices](https://github.com/gravityfargo/ckb-next#supported-devices) section links to the new .md file. If the wiki is updated using the new file, those links need to be updated as well.

tatokis commented on .github/ISSUE_TEMPLATE/new-device-request.yml:
```diff
@@ -0,0 +1,79 @@
+    attributes:
+      label: What is the name of the device?
+      description: Please be specific. Copying the name from the sticker on the device or the title of Corsair's product page will yeild the best result.
+      placeholder: ex. K95 PLATINUM XT
```
Should be: `K95 RGB PLATINUM XT`.

tatokis reviewed (changes requested):
I _absolutely love_ this.

Is it possible to leave the generated report field empty and require it to be filled in?
People not generating reports has been a real problem.
I know that they can just click in and write anything in that case, but it should at least make people have another look at it.

I think the HTML I marked should be removed as it doesn't render by default unless you click preview, and that can be confusing for people.

I also think we should have no prefixes in the issue titles, as GitHub by default places the text cursor before the text itself, so when people start writing, it ends up like

> My ckb doesn't crash enough[Enhancement]: 

We have tags for them, so it should be okay. If there isn't a tag for a specific issue type, I can create one.

Would it also be possible to move the "I need help" button below all the other options and right above the chat links, in the new issue page?

All files should end with a newline. GitHub puts that icon at the end of the files when they do not end with a newline.

Finally, please split the unrelated README changes to another PR. Once done, if easy, squash rebase your commits into a single one and force push so that it's ready to merge.

gravityfargo commented:
Changelog:
- Changed the verbiage per tatokis suggestions and hindsight commonsense. 
- Cleaned the grammatical mistakes.
- Generated reports are required, and the texts suggest a file rather than a copy/paste.
- Reverted the "Ask for Help" header on the readme to its current implementation.
- Moved the "General Information" section of the readme higher on the page.
- Fixed the "\" typo from the issue form selection page.
- Removed the prefill text for new issue titles.
- Removed the prefill HTML tags from the reports/additional information fields.
- Moved the README.md changes to a new branch to make it a separate PR.
- Removed the "Due Diligence" section from the I need help form.



Notes:
> Would it also be possible to move the "I need help" button below all the other options and right above the chat links on the new issue page?

It doesn't look like we have control of the order of forms. I've got a running list of suggestions for the form templates I'll submit to GitHub at some point.

tatokis commented:
@gravityfargo Please rebase your changes properly against master. The [github diff](https://github.com/ckb-next/ckb-next/pull/860/files) shows code changes from my commits, as well as the readme changes which are still present here.

If possible, I suggest flattening the commit history (by doing a squash rebase and force pushing) at this point as there are a lot of commits that undo changes that were meant to be temporary.

fleischie reviewed (changes requested):
I really appreciate the effort that you put into this PR!

I have made a few notes, how to improve readability and - hopefully - user interaction with the form itself. Apart from that, I would also suggest rebasing your branch onto the latest main branch and squashing the relevant pieces into coherent commits. (So one commit for issue template bug-report, one for feature-request, etc.) This would simplify our work as maintainers (I am using that term very broadly to include myself).

gravityfargo commented:
I'll get to work on making the adjustments this week. I am new to git and on the struggle bus with git. I'll take an online course this week as well to figure that out with y'alls instructions.

gravityfargo commented:
I'm doing housekeeping. I'll re-open a fresh PR soon.
